doc/media/openage/terrain_format_spec.md

```diff
@@ -0,0 +1,341 @@
+layer <id> mode=off  position=<int>
+layer <id> mode=loop position=<int> time_per_frame=<float> replay_delay=<float>
+
+# definition of a terrain frames
+# these are iterated for an animation
+frame <layer_id> <image_id> <xpos> <ypos> <xsize> <ysize> blend_id=<int>
+```
+
+
+## Data Type Formatting
+
+Type     | Example | Description
+---------|---------|---------
+int      | `5`     | Signed Integer
+float    | `1.2`   | Float
+string   | `"bla"` | String of characters enclosed by `"`
+token    | `off`   | Alphanumeric predefined keyword
+
+
+## Attributes
+
+### `version`
+
+Version of the terrain format. Increments every time the syntax
+or keywords of the format change.
+
+Parameter  | Type   | Optional | Default value
+-----------|--------|----------|--------------
+version_no | int    | No       | -
+
+**version_no**<br>
+Version number of the format.
+
+
+#### Example
+
+```
+version 1
+```
+
+
+### `imagefile`
+
+Tells the renderer which image resources it has to load.
+There has to be at least one `imagefile` defined.
+
+Parameter | Type   | Optional | Default value
+----------|--------|----------|--------------
+image_id  | int    | No       | -
+filename  | string | No       | -
+
+**image_id**<br>
+Reference ID for the resource used in this file. IDs should start at `0`.
+
+**filename**<br>
+Path to the image resource for the terrain texture on the filesystem.
+
+There are two ways to specify a path: relative and absolute. Relative
+paths are relative to the location of the terrain definition file. They
+can only refer to image resources that are in the same modpack.
+
+```
+imagefile 0 "grass.png"       # grass.png is in the same folder as the terrain file
+imagefile 1 "./grass.png"     # same as above, but more explicit
+imagefile 2 "media/sand.png"  # sand.png is in the subfolder media/
+```
+
+Absolute paths start from the (virtual) modpack root (the path where all
+modpacks are installed). They always begin with `/` followed by either
+a modpack identifier or a shortened modpack alias enclosed by `{}`. For
+information on modpack identifiers and aliases see the [modpack](modpacks.md#alias-and-identifier)
+docs.
+
+```
+imagefile 0 "/{aoe2_base@openage}/sand.png"  # absolute path with modpack identifier
+imagefile 1 "/{aoe2_base}/sand.png"          # absolute path with modpack alias
+```
+
+Absolute paths are the only way to reference image resources from other
+modpacks.
+
+
+#### Example
+
+```
+imagefile 0 "grass.png"
+imagefile 1 "../../sand.png"
+imagefile 2 "/{aoe2_base}/graphics/grass.png"
+```
+
+
+### `blendmask`
+
+Defines a blending mask that is used to blend one terrain texture
+into a different terrain texture that is directly adjacent. The
+blending mechanism is described in more detail in the [blendomatic](/doc/media/blendomatic.md)
+documentation. Only one of the blending masked defined by adjacent
+terrain is used (the one with the highest priority). Blending masks
+are **optional** and do not need to be defined.
```

Comment:
what's the content of a blendmask filename? it likely needs the same properties as tiles: texture offsets, and sizes

Reply:
It was supposed to be just the PNG with the blend tiles, but you are right. We need to define the blending mask settings somehow. I'll come up with a solution...